### PR TITLE
309/readme guidelines

### PR DESCRIPTION
**Pull request by gnayuy**
Thank you for contributing to the IPFX, your work and time will help to
advance open science! 

# Overview:
Give a brief overview of the issue you are solving. Succinctly
explain the GitHub issue you are addressing and the underlying problem
of the ticket. The commit header and body should also include this
message, for good commit messages see the [contribution guidelines](CONTRIBUTING.md).

# Addresses:
Add a link to the issue on Github board
example: 
Addresses issue [#1234](git_hub_ticket_url)

# Type of Fix:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing
      functionality to not work as expected)
- [x] Documentation Change

# Solution:
Outline your solution to the previously described issue and
underlying cause. This section should include a brief description of
your proposed solution and how it addresses the cause of the ticket

# Changes:
Include a bulleted list or check box list of the implemented changes
in brief, as well as the addition of supplementary materials (unit tests,
integration tests, etc

# Validation:
Describe how you have validated that your solution addresses the
root cause of the ticket. What have you done to ensure that your
addition is bug free and works as expected? Please provide specific
instructions so we can reproduce and list any relevant details about
your configuration

### Configuration details:

# Checklist
- [ ] My code follows
      [Allen Institute Contribution Guidelines](CONTRIBUTING.md)
- [ ] My code is unit tested and does not decrease test coverage
- [ ] I have performed a self review of my own code
- [ ] My code is well-documented, and the docstrings conform to
      [Numpy Standards](https://numpydoc.readthedocs.io/en/latest/format.html)
- [ ] I have updated the documentation of the repository where
      appropriate
- [ ] The header on my commit includes the issue number
- [ ] My code passes all tests
- [ ] I have updated the CHANGELOG.md with the description of changes understandable by end users

# Notes:
Use this section to add anything you think worth mentioning to the
reader of the issue

### CODE REVIEW

**Comment by gnayuy** [comment]
minor changes by recognizing the latest master branch changes from the last release, adding example ephys data downloading info

File: README.md [hunk 1]
```diff
@@ -10,15 +10,15 @@ IPFX is a Python package for computing intrinsic cell features from electrophysi
...
 
 Quick Start
 ------------
-To start analyzing data now, check out the [quick_start](docs/quick_start.rst) . For a more in depth guide to IPFX, see [tutorial](docs/tutorial.rst) 
+To start analyzing data now, check out the [quick_start](https://ipfx.readthedocs.io/en/latest/quick_start.html) . For a more in depth guide to IPFX, see [tutorial](https://ipfx.readthedocs.io/en/latest/tutorial.html)
 
 Contributing
 ------------
-We welcome contributions! Please see our [contribution guide](CONTRIBUTING.md) for more information. Thank you!
+We welcome contributions! Please see our [contribution guide](https://github.com/AllenInstitute/ipfx/blob/master/CONTRIBUTING.md) for more information. Thank you!
```
Comment (sgratiy) [inline README.md:22]: why did you change the links to point to the master branch? This is confusing because when clicking on links you will see content belonging to different branches.

File: docs/quick_start.rst [hunk 2]
```diff
@@ -21,3 +21,22 @@ Output:
  * output.nwb: NWB file including spike times
  * log.txt: run log
  * qc_figs: index.html includes cell figures and feature table and sweep.html includes sweep figures
+
+
```
Comment (sgratiy) [inline docs/quick_start.rst:25]: We already have a page describing how to download data on the Data Access page https://github.com/AllenInstitute/ipfx/pull/450. We should not duplicate the content to avoid needing to update two pages in the future. I believe on the quick_start.rst we should outline the steps to get started, but refer the user to the data access page for specific  download instructions. Same with the installation of ipfx. 
Also I believe, the comment in the issue #309 should be fully answered on the quick_start.rst and we would just refer the user to the updated quick_start page rather than providing quick start instructions for a particular user.

Comment (gnayuy) [reply docs/quick_start.rst:25]: remove the duplicated downloaded info and added guideline with an example as suggested.

File: README.md (hunk 1 above)
Comment (gnayuy) [reply README.md:22]: This was merged from the master branch. If I change it to this branch, after merged, will confuse other branches?

Comment (sgratiy) [reply README.md:22]: you should rebase you work on top of the release branch and target the release branch with our PR

File: docs/quick_start.rst [hunk 3]
```diff
@@ -21,3 +21,38 @@ Output:
...
+Step-by-step guideline
```
Comment (sgratiy) [inline docs/quick_start.rst:26]: I think having a separate "Step-by-step" section is confusing. We should include the guideline inside the existing quick start section. 
Something like:
To get started, install IPFX via pip into a fresh conda evironment. For details see the Installation page (link to the installation page)
Next, download some data files  according to the instructions in the Data Access (link) page.
To process a dataset saved in the nwb file, run:
...

File: docs/quick_start.rst [hunk 4]
```diff
@@ -21,3 +21,38 @@ Output:
...
+.. code-block:: bash
+
+    pip install dandi
+    dandi download https://dandiarchive.org/dandiset/000020
+
+ * pick one dataset, e.g. sub-599387254_ses-601506492_icephys.nwb from mouse data, create output directory for saving the results, for example
+
+.. code-block:: bash
+
+     mkdir outputs
```
Comment (sgratiy) [inline docs/quick_start.rst:49]: we do not need to say this. Instead the existing description on line 13 should say that it is an existing directory.

File: docs/quick_start.rst [hunk 5]
```diff
@@ -21,3 +21,38 @@ Output:
...
+     mkdir outputs/qc
```
Comment (sgratiy) [inline docs/quick_start.rst:50]: Also, I would like us to remove --qc_fig_dir parameter in favor of --save_figures that would create a relative directory automatically if save_figures=True. I created an issue https://github.com/AllenInstitute/ipfx/issues/454

File: docs/quick_start.rst (hunk 4 above)
Comment (gnayuy) [reply docs/quick_start.rst]: removed, thanks

File: docs/quick_start.rst (hunk 5 above)
Comment (gnayuy) [reply docs/quick_start.rst]: removed it as well, thanks.

File: docs/quick_start.rst (hunk 3 above)
Comment (gnayuy) [reply docs/quick_start.rst]: modified as suggested, thanks.

File: docs/quick_start.rst [hunk 6]
```diff
@@ -21,3 +24,4 @@ Output:
  * output.nwb: NWB file including spike times
```
Comment (sgratiy) [inline docs/quick_start.rst:24]: minor: I have noticed that the latest code instead creates input_json and output_json. @NileGraddis. Could we may be make them input.json and output.json instead (i.e., to have an extension .json)? In any case, we need the documentation to match the JSON file names.
